draft: build on BTC dev docs (#5697)
dfinity/portal: docs/build-on-btc/btc-api.mdx

---
keywords: [advanced, bitcoin, bitcoin api, api reference, api endpoints, bitcoin reference, bitcoin endpoints, btc api]
hide_table_of_contents: true
---

import TabItem from "@theme/TabItem";
import { AdornedTabs } from "/src/components/Tabs/AdornedTabs";
import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";

# Bitcoin API endpoints

<MarkdownChipRow labels={["Bitcoin"]} />

The Bitcoin API is hosted as an ICP smart contract. To make calls to the API, you must target the smart contract address for either the Bitcoin mainnet or Bitcoin testnet:

- Mainnet: [`ghsi2-tqaaa-aaaan-aaaca-cai`](https://dashboard.internetcomputer.org/canister/ghsi2-tqaaa-aaaan-aaaca-cai)

- Testnet (v4): [`g4xu7-jiaaa-aaaan-aaaaq-cai`](https://dashboard.internetcomputer.org/canister/g4xu7-jiaaa-aaaan-aaaaq-cai)

[View the full Bitcoin API specification for details.](https://github.com/dfinity/bitcoin-canister/blob/master/INTERFACE_SPECIFICATION.md)

## Available endpoints

### `bitcoin_get_utxos`

> Returns UTXOs for a Bitcoin address. UTXOs can be filtered by minimum confirmations (`min_confirmations`, max 144) or via `page` token.

Reviewer: ```suggestion
Returns UTXOs for a Bitcoin address. UTXOs can be filtered by minimum confirmations (`min_confirmations`, which may be at most 144) or via a `page` reference.
```
```suggestion
Returns UTXOs for a Bitcoin address. UTXOs can be filtered by minimum confirmations (`min_confirmations`, max 144) or via `page` token.
```

### `bitcoin_get_utxos_query`

> Used to query `bitcoin_get_utxos`. Since this is a query call, it is returned quickly but results should not be trusted for secure applications.

Reviewer: ```suggestion
Used to query `bitcoin_get_utxos`. Since this is a query call, it returns quickly but results are not trustworthy.
```
```suggestion
Used to query `bitcoin_get_utxos`. Since this is a query call, it is returned quickly but results should not be trusted for secure applications.
```

> Used to query `bitcoin_get_utxos`. Since this is a query call, it is returned quickly but results should not be trusted for secure applications.

Reviewer: ```suggestion
Used to query `bitcoin_get_utxos`. Since this is a query call, it is returns quickly but results are not trusthworthy.
```
